Cython/Utility/Optimize.c

```diff
@@ -1181,8 +1181,13 @@ static {{c_ret_type}} {{cfunc_name}}(PyObject *op1, PyObject *op2, long intval,
             return PyLong_Type.tp_as_number->nb_{{slot_name}}(op1, op2);
         {{elif c_op == '%'}}
             // see CMath.c :: ModInt utility code
-            x = a % b;
-            x += ((x != 0) & ((x ^ b) < 0)) * b;
+            if (unlikely((b & (b - 1)) == 0)) {
```

Comment:
I wonder if `unlikely` is a mistake here. It's probably relatively common. We mainly use `unlikely` for error handling (where we expect it to be the slow path).

Not smart enough right now to know if it's correct :)